data/transaction/status.go

```diff
@@ -70,3 +80,33 @@ func (params *StatusComputer) isDestinationMe() bool {
 func (params *StatusComputer) isContractDeploy() bool {
 	return core.IsEmptyAddress(params.Receiver) && len(params.TransactionData) > 0
 }
+
+// SetStatusIfIsRewardReverted will compute and set status for a reverted reward transaction
+func (params *StatusComputer) SetStatusIfIsRewardReverted(tx *ApiTransactionResult) bool {
+	if params.MiniblockType != block.RewardsBlock {
+		return false
+	}
+
+	var storerUnit dataRetriever.UnitType
+
+	selfShardID := params.SelfShard
+	if selfShardID == core.MetachainShardId {
+		storerUnit = dataRetriever.MetaHdrNonceHashDataUnit
+	} else {
+		storerUnit = dataRetriever.ShardHdrNonceHashDataUnit + dataRetriever.UnitType(selfShardID)
+	}
+
+	nonceToByteSlice := params.Uint64ByteSliceConverter.ToByteSlice(params.HeaderNonce)
+	headerHash, err := params.Store.Get(storerUnit, nonceToByteSlice)
+	if err != nil {
+		// this should never happen
```

Comment:
please add a log.Error here and remove comment. BTW, it can happen if we encounter disk errors.

Reply:
done.